backend/src/api/users.ts

```diff
@@ -22,6 +28,64 @@ import {
   isOrgAdmin,
   isGlobalWriteAdmin
 } from './auth';
+import { Type, plainToClass } from 'class-transformer';
+import { IsNull } from 'typeorm';
+
+class UserSearch {
+  @IsInt()
+  @IsPositive()
+  page: number = 1;
+
+  @IsString()
+  @IsIn([
+    'fullName',
+    'firstName',
+    'lastName',
+    'email',
+    'name',
+    'userType',
+    'dateAcceptedTerms',
+    'lastLoggedIn',
+    'acceptedTermsVersion'
+  ])
+  @IsOptional()
+  sort: string = 'fullName';
+
+  @IsString()
+  @IsIn(['ASC', 'DESC'])
+  order: 'ASC' | 'DESC' = 'DESC';
+
+  @IsInt()
+  @IsOptional()
+  // If set to -1, returns all results.
+  pageSize?: number;
+
+  @IsString()
+  @IsOptional()
+  @IsIn(['title'])
+  groupBy?: 'title';
+
+  async getResults(event): Promise<[User[], number]> {
+    const pageSize = this.pageSize || 25;
+    const sort =
+      this.sort === 'name' || 'fullName'
+        ? 'user.fullName'
+        : this.sort === 'userType'
```

Comment:
Instead of these ternary statements, would it be possible just to do `user.${this.sort}`? Or would there be an issue potentially with that?